### PR TITLE
Issue 2/units and quality kinds

### PR DESCRIPTION
**Pull request by MuazzamChaud**

### CODE REVIEW

File: scripts/populatejsonfiles.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    """
+
+    IS_A_QUDT_QUANTITY_KIND = """
+    SELECT ?url {
+        ?url a <http://qudt.org/schema/qudt/QuantityKind> 
+    }
+    """
+
+
+class Resource(NamedTuple):
```
Comment (roblinksdata) [inline scripts/populatejsonfiles.py]: I'd say dataclasses are better than NamedTuples @mikeAdamss .

File: scripts/populatejsonfiles.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    """
+    Function to update the enum urls in a json file with any urls from the input
+    list that are not already present.
+    """
+
+    with open(json_path, "r") as jsonFile:
+        data = json.load(jsonFile)
+
+        for url in url_list:
+            if url not in data["uris"]["enum"]:
```
Comment (roblinksdata) [inline scripts/populatejsonfiles.py:87]: I think this is going to get quite large, can you convert it into a set before going through this for-loop and check whether url is in that? That should make it speedier as we build everything up.

I was going to suggest we delete values we don't find in the ttl but I guess we should always be adding, so I think you've made the right choice there.

File: scripts/tests/unit/test_populatejsonfiles.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    with pytest.raises(HTTPError) as excinfo:
+        r = get(url)
+        assert f'Could not get "{url}" with status code {r.status_code}' in str(excinfo)
+
+@pytest.mark.vcr
+def test_update_json():
+    """
+    test update json does not add duplicates
+    """
+    json_path = '/workspaces/reusable-rdf-resources/quantity-kinds.json'
```
Comment (roblinksdata) [inline scripts/tests/unit/test_populatejsonfiles.py]: Can you run these unit tests inside a temporary directory so we don't end up writing over the production files?

File: scripts/tests/unit/test_populatejsonfiles.py [hunk 4]
```diff
@@ -39,25 +41,27 @@ def test_update_json():
...
-    json_path = '/workspaces/reusable-rdf-resources/quantity-kinds.json'
-    url_list = ("http://qudt.org/vocab/quantitykind/AbsoluteActivity",
-      "http://qudt.org/vocab/quantitykind/AbsoluteHumidity",
-      "http://qudt.org/vocab/quantitykind/AbsorbedDose",
-      "http://qudt.org/vocab/quantitykind/AbsorbedDoseRate",
-      "http://qudt.org/vocab/quantitykind/Absorptance",
-      "http://qudt.org/vocab/quantitykind/Acceleration",)
+    with TemporaryDirectory() as temp_dir:
+        temp_dir = Path(temp_dir)
+        json_path = temp_dir / '/workspaces/reusable-rdf-resources/quantity-kinds.json'
```
Comment (roblinksdata) [inline scripts/tests/unit/test_populatejsonfiles.py]: How does this work without copying the file over to the temp dir first?

**Review by roblinksdata — APPROVED** [review]

File: scripts/populatejsonfiles.py (hunk 1 above)
Comment (mikeAdamss) [reply scripts/populatejsonfiles.py]: yeah fair enough, old habits 😄

**Pull request merged**